### PR TITLE
Use `getClientAddress` instead of `getClientIP`.

### LINKED ISSUE #11142: Replace use of `getClientIP` with `getClientAddress`
Running unit tests I noticed lots of deprecation warnings:

```
DeprecationWarning: twisted.web.http.Request.getClientIP was deprecated in Twisted 18.4.0; please use getClientAddress instead
  tags.PEER_HOST_IPV6: request.getClientIP(),
```

We are using Twisted >= 18.9:

https://github.com/matrix-org/synapse/blob/40f619eaa54d2391deccec473fc0f655c379e766/synapse/python_dependencies.py#L64

So it should be safe to use the nondeprecated alternative.

The difference [seems to be](https://twistedmatrix.com/documents/21.2.0/api/twisted.web.http.Request.html#getClientIP) getClientIp returns a `str` whereas `getClientAddress` returns an `IAddress`. The documentation says:

> Callers must check the type of the returned address.

which is a bit of a faff. It's probably safe to assume that every where we call it we either get an `IPv4Address` or `IPv6Address`, both of which have a `host` attribute which is probably what we want. I think this needs a little more thought to convince ourselves that's right. But once we do that I'm pretty sure fixing these up will be routine.

### PR DESCRIPTION
**Pull request by clokep**
`getClientIP` is deprecated, switch to `getClientAddress`.

Fixes #11142

### CODE REVIEW

File: tests/replication/_base.py [hunk 1]
```diff
@@ -154,10 +154,12 @@ def handle_http_replication_attempt(self) -> SynapseRequest:
         self.assertEqual(port, 8765)
 
         # Set up client side protocol
-        client_protocol = client_factory.buildProtocol(None)
+        client_address = IPv4Address("TCP", "127.0.0.1", 1234)
+        client_protocol = client_factory.buildProtocol(client_address)
```
Comment (H-Shay) [inline tests/replication/_base.py:158]: Just curious why the change from calling `buildProtocol` with `client_address` as the parameter vs `None`?

Comment (clokep) [reply tests/replication/_base.py:158]: The address is supposed to be passed, `None` is not a legal value there: https://twistedmatrix.com/documents/current/api/twisted.internet.interfaces.IProtocolFactory.html#buildProtocol

Although now that I check it is supposed to be a tuple version, not an `IAddress` instance. 😢

Comment (clokep) [reply tests/replication/_base.py:158]: > Was it just not working then, or did it work but you're changing it to be more consistent with the proper usage?

I think (currently) the value never gets used, I was hoping this would fix the bug, but the "real" fix was adding it to the transport. It seemed best to leave it as it was more correct to pass a value here than `None`.

Comment (H-Shay) [reply tests/replication/_base.py]: Cool thanks for answering!

**Review by H-Shay — APPROVED** [review]

File: tests/replication/_base.py (hunk 1 above)
Comment (H-Shay) [reply tests/replication/_base.py]: LGTM assuming tests pass

Comment (clokep) [reply tests/replication/_base.py]: > Cool thanks for answering!

No problem!

I think the tl;dr is "It was wrong, but fixing it doesn't fix it. Let's include the fix anyway" 😆
